queue/queue.go

```diff
@@ -0,0 +1,278 @@
+			conn,
+			makeCmdMap(name),
+		}
+	} else {
+		err = fmt.Errorf("Tube %s does not exist", name)
+	}
+
+	return q, err
+}
+
+// Put data to queue. Returns task.
+func (q queue) Put(data interface{}) (*Task, error) {
+	return q.put(data)
+}
+
+// Put data with options (ttl/ttr/pri/delay) to queue. Returns task.
+func (q queue) PutWithOpts(data interface{}, cfg Opts) (*Task, error) {
+	return q.put(data, cfg.toMap())
+}
+
+func (q queue) put(params ...interface{}) (*Task, error) {
+	var task *Task
+	resp, err := q.conn.Call(q.cmd["put"], params)
+	if err == nil {
+		task, err = toTask(resp.Data, &q)
+	}
+
+	return task, err
+}
+
+// The take request searches for a task in the queue.
+func (q queue) Take() (*Task, error) {
+	var params interface{}
+	if q.conn.ConfiguredTimeout() > 0 {
+		params = q.conn.ConfiguredTimeout().Seconds()
+	}
+	return q.take(params)
+}
+
+// The take request searches for a task in the queue. Waits until a task becomes ready or the timeout expires.
+func (q queue) TakeTimeout(timeout time.Duration) (*Task, error) {
+	t := q.conn.ConfiguredTimeout()
+	if t > 0 && timeout > t {
+		timeout = t
+	}
+	return q.take(timeout.Seconds())
+}
+
+func (q queue) take(params interface{}) (*Task, error) {
+	var t *Task
+	resp, err := q.conn.Call(q.cmd["take"], []interface{}{params})
+	if err == nil {
+		t, err = toTask(resp.Data, &q)
+	}
+	return t, err
+}
+
+// Drop queue.
+func (q queue) Drop() error {
+	_, err := q.conn.Call(q.cmd["drop"], []interface{}{})
+	return err
+}
+
+// Look at a task without changing its state.
+func (q queue) Peek(taskId uint64) (*Task, error) {
+	resp, err := q.conn.Call(q.cmd["peek"], []interface{}{taskId})
+	if err != nil {
+		return nil, err
+	}
+
+	t, err := toTask(resp.Data, &q)
+
+	return t, err
+}
+
+func (q queue) _ack(taskId uint64) (string, error) {
+	return q.produce("ack", taskId)
+}
+
+func (q queue) _delete(taskId uint64) (string, error) {
+	return q.produce("delete", taskId)
+}
+
+func (q queue) _bury(taskId uint64) (string, error) {
+	return q.produce("bury", taskId)
+}
+
+func (q queue) _release(taskId uint64, cfg Opts) (string, error) {
+	return q.produce("release", taskId, cfg.toMap())
+}
+func (q queue) produce(cmd string, params ...interface{}) (string, error) {
+	resp, err := q.conn.Call(q.cmd[cmd], params)
+	if err != nil {
+		return "", err
+	}
+
+	t, err := toTask(resp.Data, &q)
+	if err != nil {
+		return "", err
+	}
+	return t.status, nil
+}
+
+// Reverse the effect of a bury request on one or more tasks.
+func (q queue) Kick(count uint64) (uint64, error) {
+	resp, err := q.conn.Call(q.cmd["kick"], []interface{}{count})
+	var id uint64
+	if err == nil {
+		id = resp.Data[0].([]interface{})[0].(uint64)
+	}
+	return id, err
+}
+
+// Return the number of tasks in a queue broken down by task_state, and the number of requests broken down by the type of request.
+func (q queue) Statistic() (interface{}, error) {
+	resp, err := q.conn.Call(q.cmd["statistics"], []interface{}{})
```

Comment:
Pass the name of a queue as a parameter.
Otherwise you will got statistic for all queues.